drivers/SmartThings/deepsmart/src/deepsmart/wisers.lua

```diff
@@ -338,30 +334,30 @@ function Wisers.add_wisers(wisers, is_add)
       wiser["dpenum"] = dpenum.load_config(v.uuid, nil)
       wiser.state = 1
       wiser.ctrlmap = {}
+      -- wiser loop for get updated devices from wiser
+      -- every 2second use https api to get updated devices from given time
+      -- after get results save the last time for next request
       Wisers.driver:call_on_schedule(
       2,
       function ()
         return Wisers.wiser_loop(v.uuid)
       end,
       'wiser loop')
-      --cosock.spawn(function() Wisers.wiser_loop(v.uuid) end, "Wiser loop")
     end
 
     -- create wiser local storage
     if (Wisers.driver.datastore[v.uuid] == nil) then
       Wisers.driver.datastore[v.uuid] = {}
     end
     Wisers.driver.datastore[v.uuid].ip = v.ip
-    Wisers.driver.datastore[v.uuid].port = v.port
     Wisers.driver.datastore[v.uuid].wiser_index_code = v.uuid
     wiser["indexCode"] = v.uuid
     wiser["wiser_index_code"] = v.uuid
     wiser["last_active_time"] = "0"
     -- deepsmart wiser id is 8bytes integer
     wiser["wiser_bytes"] = utils.tonumber(v.uuid)
     wiser["ip"] = v.ip
-    wiser["port"] = v.port
-    wiser["api"] = api.client(v.uuid, v.ip)
+    --wiser["api"] = api.client(v.uuid, v.ip)
```

Comment:
whats the reason for re-creating the rest client when needed rather than storing it on the object? With some of the repeated calls to get updated devices every 2 seconds, I would think this is something we want to keep around rather than pay the penalty of re-creating and then garbage collecting the tables all the time.